qiskit/extensions/quantum_initializer/initializer.py

```diff
@@ -13,380 +13,85 @@
-            circuit.append(target_gate(list_of_angles[0]), [q[0]])
-            return circuit
-
-        # calc angle weights, assuming recursion (that is the lower-level
-        # requested angles have been correctly implemented by recursion
-        angle_weight = np.kron([[0.5, 0.5], [0.5, -0.5]], np.identity(2 ** (local_num_qubits - 2)))
-
-        # calc the combo angles
-        list_of_angles = angle_weight.dot(np.array(list_of_angles)).tolist()
-
-        # recursive step on half the angles fulfilling the above assumption
-        multiplex_1 = self._multiplex(target_gate, list_of_angles[0 : (list_len // 2)], False)
-        circuit.append(multiplex_1.to_instruction(), q[0:-1])
-
-        # attach CNOT as follows, thereby flipping the LSB qubit
-        circuit.append(CXGate(), [msb, lsb])
-
-        # implement extra efficiency from the paper of cancelling adjacent
-        # CNOTs (by leaving out last CNOT and reversing (NOT inverting) the
-        # second lower-level multiplex)
-        multiplex_2 = self._multiplex(target_gate, list_of_angles[(list_len // 2) :], False)
-        if list_len > 1:
-            circuit.append(multiplex_2.to_instruction().reverse_ops(), q[0:-1])
-        else:
-            circuit.append(multiplex_2.to_instruction(), q[0:-1])
-
-        # attach a final CNOT
-        if last_cnot:
-            circuit.append(CXGate(), [msb, lsb])
-
-        return circuit
-
-    def broadcast_arguments(self, qargs, cargs):
-        flat_qargs = [qarg for sublist in qargs for qarg in sublist]
-
-        if self.num_qubits != len(flat_qargs):
-            raise QiskitError(
-                "Initialize parameter vector has %d elements, therefore expects %s "
-                "qubits. However, %s were provided."
-                % (2**self.num_qubits, self.num_qubits, len(flat_qargs))
-            )
-        yield flat_qargs, []
-
-    def validate_parameter(self, parameter):
-        """Initialize instruction parameter can be str, int, float, and complex."""
-
-        # Initialize instruction parameter can be str
-        if self._from_label:
-            if parameter in ["0", "1", "+", "-", "l", "r"]:
-                return parameter
-            raise CircuitError(
-                "invalid param label {} for instruction {}. Label should be "
-                "0, 1, +, -, l, or r ".format(type(parameter), self.name)
-            )
-
-        # Initialize instruction parameter can be int, float, and complex.
-        if isinstance(parameter, (int, float, complex)):
-            return complex(parameter)
-        elif isinstance(parameter, np.number):
-            return complex(parameter.item())
-        else:
-            raise CircuitError(f"invalid param type {type(parameter)} for instruction  {self.name}")
+        return self._stateprep.gates_to_uncompute()
 
 
 def initialize(self, params, qubits=None):
-    r"""Initialize qubits in a specific state.
+    """Initialize qubits in a specific state.
```

Comment:
is the "r" needed here because we're using `\ket{0}` below? I thought the "raw" string is always needed when you're using the escape sequence `\` but don't want to escape and instead use it e.g. for latex math

Reply:
yeah CI failed bc of my kets 😅 I assumed it worked bc it was fine when i built the docs locally, I'll change them